### PR TITLE
Add initial device snmp test suite

### PR DESCRIPTION
**Pull request by MonicaisHer**
This PR will solve: https://github.com/canonical/edgex-snap-testing/issues/30

### CODE REVIEW

File: test/suites/device-snmp/main_test.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	// install the device snap before edgexfoundry
+	// to catch build error sooner and stop
+	if utils.LocalSnap != "" {
+		utils.SnapInstallFromFile(nil, utils.LocalSnap)
+	} else {
+		utils.SnapInstallFromStore(nil, deviceSnmpSnap, utils.ServiceChannel)
+	}
+	utils.SnapInstallFromStore(nil, "edgexfoundry", utils.PlatformChannel)
+
+	// make sure all services online before starting subtests
```
Comment (farshidtz) [inline test/suites/device-snmp/main_test.go:43]: Note that `Test*` are tests and `t.Run("*",..` inside those tests are subtests. There could also be sub-subtests i.e. t.Run inside another t.Run.

**Review by farshidtz — APPROVED** [review]
Looks good. Thanks.

**Review by farshidtz — CHANGES_REQUESTED** [review]

File: test/utils/snap.go [hunk 2]
```diff
@@ -133,3 +133,21 @@ func SnapRestart(t *testing.T, names ...string) {
 		))
 	}
 }
+
+func SnapSetUnset(t *testing.T, snapName, defaultServicePort string) {
```
[the remark this answers is not in the document]
Comment (farshidtz) [reply test/utils/snap.go:137]: Also, the cleanup must unset. Otherwise, in case something between set and unset fails, the option will remain.

**Comment by MonicaisHer** [comment]
Could we increase the default timeout from [11m](https://github.com/canonical/edgex-snap-testing/runs/5833828464?check_suite_focus=true#step:4:121) to 30min [this place](https://github.com/canonical/edgex-snap-testing/blob/main/test/action.yml#L47)?

https://stackoverflow.com/questions/27778280/test-killed-with-quit-ran-too-long

File: test/utils/snap.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (farshidtz) [reply test/utils/snap.go]: See https://github.com/canonical/edgex-snap-testing/pull/32/commits/882fab603b0b70bc39d4e2b402b77dcce9c330d0

**Review by farshidtz — APPROVED** [review]

**Pull request merged**